manifests/init.pp

```diff
@@ -164,6 +165,7 @@
   Boolean $enable_views                                             = $dns::params::enable_views,
   Hash[String, Hash] $zones                                         = $dns::params::zones,
   Hash[String, Hash] $keys                                          = $dns::params::keys,
+  Optional[Hash] $logging                                           = {},
```

Comment:
Would it make sense to add a a type for this to document/verify the passed in datastructure?

Reply:
I converted this to being implemented as 2 different defined types.  If it looks reasonable, I will squash/rebase this branch.

Reply:
@ekohl This was a better solution, so I squashed and rebased.